File: src/InvertedIndex.java
import java.util.*;

public class InvertedIndex {

    private final HashMap<String, Set<String>> invertedIndex = new HashMap<>();

    public void buildInvertedIndex(Map<String, String> DocNameMapToContent) {
        for (Map.Entry<String, String> doc : DocNameMapToContent.entrySet()) {
            String[] words = splitDocumentsWords(doc.getValue());
            addDocumentWords(doc.getKey(), words);
        }
    }

    private String[] splitDocumentsWords(String docContent) {
        return docContent.split("[\\W]+");
    }

    private void addDocumentWords(String docName, String[] docWords) {
        for (String word : docWords) {
            if(invertedIndex.containsKey(word))
                invertedIndex.get(word).add(docName);
            else
                invertedIndex.put(word, new HashSet<>(Arrays.asList(docName)));
        }
    }

    public Set<String> getInvertedIndexValue(String key) {
        return invertedIndex.getOrDefault(key, new HashSet<>());
    }

}

File: Phase05/Phase05/Phase05/InvertedIndex.cs
using System.Collections.Generic;
using System.Text.RegularExpressions;

namespace Phase05
{
    public class InvertedIndex : IInvertedIndex
    {
        private Dictionary<string, HashSet<string>> _invertedIndexMap = new Dictionary<string, HashSet<string>>();

        public void BuildInvertedIndex(Dictionary<string, string> docMapToContent)
        {
            foreach (var doc in docMapToContent)
            {
                var words = SplitDocumentsWords(doc.Value);
                AddDocumentWords(doc.Key, words);
            }
        }

        private string[] SplitDocumentsWords(string docContent)
        {
            return Regex.Split(docContent,"[\\W]+");
        }

        private void AddDocumentWords(string docName, string[] docWords)
        {
            foreach (var word in docWords)
            {
                if (_invertedIndexMap.ContainsKey(word))
                    _invertedIndexMap[word].Add(docName);
                else
                    _invertedIndexMap.Add(word, new HashSet<string> {docName});
            }
        }


        public HashSet<string> GetValueOfInvertedIndexKey(string key)
        {
            return _invertedIndexMap.GetValueOrDefault(key, new HashSet<string>());
        }
    }
}